crates/miden-lib/src/transaction/mod.rs

```diff
@@ -433,22 +439,30 @@ impl TransactionKernel {
     const KERNEL_TESTING_LIB_BYTES: &'static [u8] =
         include_bytes!(concat!(env!("OUT_DIR"), "/assets/kernels/kernel_library.masl"));
 
-    pub fn kernel_as_library() -> miden_objects::assembly::Library {
-        miden_objects::assembly::Library::read_from_bytes(Self::KERNEL_TESTING_LIB_BYTES)
+    /// Returns the kernel library.
+    pub fn library() -> Library {
+        Library::read_from_bytes(Self::KERNEL_TESTING_LIB_BYTES)
             .expect("failed to deserialize transaction kernel library")
     }
 
+    /// Returns the mock account and faucet libraries.
+    pub fn mock_libraries() -> impl Iterator<Item = Library> {
+        use miden_objects::account::AccountCode;
+
+        use crate::testing::mock_account_code::MockAccountCodeExt;
+
+        vec![AccountCode::mock_account_library(), AccountCode::mock_faucet_library()].into_iter()
+    }
+
     /// Returns an [`Assembler`] with the transaction kernel as a library.
     ///
     /// This assembler is the same as [`TransactionKernel::assembler`] but additionally includes the
     /// kernel library on the namespace of `$kernel`. The `$kernel` library is added separately
     /// because even though the library (`api.masm`) and the kernel binary (`main.masm`) include
     /// this code, it is not otherwise accessible. By adding it separately, we can invoke procedures
     /// from the kernel library to test them individually.
-    pub fn with_kernel_library() -> Assembler {
-        let source_manager: Arc<dyn SourceManager + Send + Sync> =
-            Arc::new(DefaultSourceManager::default());
-        let kernel_library = Self::kernel_as_library();
+    pub fn with_kernel_library(source_manager: Arc<dyn SourceManagerSync>) -> Assembler {
+        let kernel_library = Self::library();
 
         #[cfg(all(any(feature = "testing", test), feature = "std"))]
         source_manager_ext::load_masm_source_files(&source_manager);
```

Comment:
Is there a reason not to call `Sellf:assembler_with_source_manager()` here and then add the extra libraries to the returned assembler? This way, we won't need to call `source_manager_ext::load_masm_source_files(&source_manager)`.

Or if do need to call it, we could probably get rid of `any(feature = "testing", test)` here since we are already in a code block that is gated by these features.